pygem/params.py

```diff
@@ -294,6 +299,79 @@ def print_info(self):
 		print '\nposition_vertex_3 ='
 		print self.position_vertex_3
 
+	def build_bounding_box(self, shape, tol=1e-6, triangualte=False, triangulate_tol=1e-1):
+		min_xyz, max_xyz = self._calculate_bb_dimension(shape, tol, triangualte, triangulate_tol)
+		self.origin_box = min_xyz
+		self._set_box_dimensions(min_xyz, max_xyz)
+		self._set_position_of_vertices()
+		self._set_mapping()
+		self._set_transformation_params_to_zero()
+
+	def _set_box_origin(self, xyz):
+		self.origin_box = xyz
+
+	def _set_box_dimensions(self, min_xyz, max_xyz):
+		"""
+		Dimensions of the cage are set as distance from the origin (minimum) of the cage to
+		the maximal point in each dimension.
+		:return:
+		"""
+		dims = [max_xyz[i] - min_xyz[i] for i in range(3)]
+		self.lenght_box_x = dims[0]
+		self.lenght_box_y = dims[1]
+		self.lenght_box_z = dims[2]
+
+	def _set_position_of_vertices(self):
+		"""
+		Vertices of the control box around the object are set in this method.
+		Four vertex (non coplanar) are sufficient to uniquely identify a parallelepiped -- the
+		second half of the box is created as a mirror reflection of the first four vertices.
+		:return:
+		"""
+		origin_array = np.array(self.origin_box)
+		dim = [self.lenght_box_x, self.lenght_box_y, self.lenght_box_z]
+		self.position_vertex_0 = origin_array
+		self.position_vertex_1 = origin_array + np.array([dim[0], .0, .0])
+		self.position_vertex_2 = origin_array + np.array([.0, dim[1], .0])
+		self.position_vertex_3 = origin_array + np.array([.0, .0, dim[2]])
+
+	def _set_mapping(self):
+		dim = [self.lenght_box_x, self.lenght_box_y, self.lenght_box_z]
+		self.psi_mapping = np.diag([1. / dim[i] for i in range(3)])
+		self.inv_psi_mapping = np.diag(dim)
+
+	def _set_transformation_params_to_zero(self):
+		ctrl_pnts = self.n_control_points
+		self.array_mu_x = np.zeros(ctrl_pnts)
+		self.array_mu_y = np.zeros(ctrl_pnts)
+		self.array_mu_z = np.zeros(ctrl_pnts)
+
+	def _calculate_bb_dimension(self, shape, tol=1e-6, triangualte=False, triangulate_tol=1e-1):
+		""" return the bounding box of the TopoDS_Shape `shape`
+		Parameters
```

Comment:
I am not sure that documenting like this will produce the same result as using :param etc. Please check and fix accordingly.